app/main/views/index.py

```diff
@@ -19,12 +19,22 @@ def index():
     if subscription_form.validate_on_submit():
         return redirect(url_for('main.subscription', email=subscription_form.email.data))
 
+    all_events = future_events
+    if len(all_events) < 3:
+        past_events = api_client.get_events_past_year()
+
+        while len(all_events) < 3:
+            event = past_events.pop(-1)
+            event['past'] = True
+            all_events.append(event)
+
     return render_template(
         'views/home.html',
         images_url=current_app.config['IMAGES_URL'],
         main_article=articles[index],
         articles=articles,
-        events=events,
+        all_events=all_events,
+        future_events=future_events,
```

Comment:
I don't think you need to pass `future_events` through, it should be possible to check on the template if an event is a past event in which case ignore on the carousel. You might also want to put a conditional check if there are future events so show the carousel or not.